rsconnect/main.py

```diff
@@ -244,6 +244,58 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+# inverts bool args if they are provided, otherwise returns None
+def env_management_callback(ctx, param, value):
+    # eval the shorthand flag if it was provided
+    disable_env_management = ctx.params.get('disable_env_management')
+    if disable_env_management is not None:
+        value = disable_env_management
+
+    # invert env_management value if it was provided
+    if value is not None:
+        return not value
+    return value
+
+
+def runtime_environment_args(func):
+    @click.option(
+        "--image",
+        "-I",
+        help="Target image to be used during content build and execution. "
+        "This option is only applicable if the Connect server is configured to use off-host execution.",
+    )
+    @click.option(
+        "--disable-env-management",
+        is_flag=True,
+        is_eager=True,
+        default=None,
+        help="Shorthand to disable environment management for both Python and R.",
+    )
+    @click.option(
+        "--disable-env-management-py",
+        "env_management_py",
+        is_flag=True,
+        default=None,
+        help="Disable Python environment management. Connect will not perform any Python package installation. "
+        "It is the user's responsibility to ensure required packages are installed in the runtime environment.",
```

Comment:
I think there are parts of this which might be unclear to end users. Going to think through it in this comment.

> Disable Python environment management. Connect will not perform any Python package installation. It is the user's responsibility to ensure required packages are installed in the runtime environment.

- "the user's responsibility" — actually probably the server admin's responsibility
- "the runtime environment" — might be unclear for users without a clear mental model of the runtime environment on the Connect server that we're talking about, who might think this is referring to their local runtime environment.

Maybe something like 

> Disable Python environment management for this bundle. Connect will not create an environment or instal packages. An administrator must install any required packages in the correct Python environment on the Connect server.

And similar for the R help.

Reply:
I like this; I think the "for this bundle" distinction is also really important. Since the setting in the `manifest.json`, and thus the bundle, have the highest precedence it makes sense to explicitly say that these changes must be applied to _every bundle_ for them to take effect.  

If the user wants to apply these changes to all bundles then they'll need to use the server API to set an environment management strategy at the Application-level.